archive/tar_stream.ts

```diff
@@ -0,0 +1,466 @@
+   *   .pipeTo((await Deno.create('./out.tar.gz')).writable)
+   * ```
+   */
+  get writable(): WritableStream<TarStreamInput> {
+    return this.#writable;
+  }
+}
+
+function parsePath(
+  path: string,
+): [Uint8Array, Uint8Array] {
+  const name = new TextEncoder().encode(path);
+  if (name.length <= 100) {
+    return [new Uint8Array(0), name];
+  }
+
+  if (name.length > 256) {
+    throw new RangeError("Path cannot exceed 256 bytes");
+  }
+
+  // If length of last part is > 100, then there's no possible answer to split the path
+  let suitableSlashPos = Math.max(0, name.lastIndexOf(SLASH_CODE_POINT)); // always holds position of '/'
+  if (name.length - suitableSlashPos > 100) {
+    throw new RangeError("Filename cannot exceed 100 bytes");
+  }
+
+  for (
+    let nextPos = suitableSlashPos;
+    nextPos > 0;
+    suitableSlashPos = nextPos
+  ) {
+    // disclaimer: '/' won't appear at pos 0, so nextPos always be > 0 or = -1
+    nextPos = name.lastIndexOf(SLASH_CODE_POINT, suitableSlashPos - 1);
+    // disclaimer: since name.length > 100 in this case, if nextPos = -1, name.length - nextPos will also > 100
+    if (name.length - nextPos > 100) {
+      break;
+    }
+  }
+
+  const prefix = name.slice(0, suitableSlashPos);
+  if (prefix.length > 155) {
+    throw new Error(
+      "Path needs to be split-able on a forward slash separator into [155, 100] bytes respectively",
+    );
+  }
+  return [prefix, name.slice(suitableSlashPos + 1)];
+}
+
+/**
+ * The type that may be returned from the `validPath` function.
+ */
+export type PathResult = { ok: true } | { ok: false; error: string };
+
+/**
+ * validPath is a function that validates the correctness of a path that
+ * may be piped to a `TarStream`. It provides a means to check that a path is
+ * valid before pipping it through the `TarStream`, where if invalid will throw an
+ * error. Ruining any progress made when archiving.
+ *
+ * @param path The path as a string
+ * @return { ok: true } | { ok: false, error: string }
+ *
+ * @example Usage
+ * ```ts
+ * import { validPath, type PathResult } from "@std/archive";
+ * import { assertEquals } from "@std/assert";
+ *
+ * assertEquals<PathResult>(validPath('MyFile.txt'), { ok: true });
+ * ```
+ */
+export function validPath(path: string): PathResult {
+  try {
+    parsePath(path);
+    return { ok: true };
+  } catch (error) {
+    if (!(error instanceof Error)) throw error;
+    return { ok: false, error: error.message };
+  }
+}
+
+/**
+ * validTarStreamOptions is a function that returns a true if all of the options
+ * provided are in the correct format, otherwise returns false.
+ *
+ * @param options The TarStreamOptions
+ * @return boolean
+ *
+ * @example Usage
+ * ```ts
+ * import { validTarStreamOptions } from "@std/archive";
+ * import { assertEquals } from "@std/assert";
+ *
+ * assertEquals(validTarStreamOptions({ mode: 755 }), true);
+ * ```
+ */
+export function validTarStreamOptions(
```

Comment:
Ditto. Can you please make this an assertion function? Perhaps, we can improve on errors in a future iteration.